sql-cli/sql_cli/configuration.py

```diff
@@ -88,34 +95,26 @@ def write_value_to_yaml(self, section: str, key: str, value: str) -> None:
         :param key: Key within the YAML file associated to the value to be recorded.
         :param value: Value associated to the key in the YAML file.
         """
-        yaml_config = self.from_yaml_to_dict()
+        yaml_config = self.from_yaml_to_dict(filepath)
         yaml_config.setdefault(section, {})
         yaml_config[section][key] = value
 
-        filepath = self.get_filepath()
-        with open(filepath, "w") as fp:
+        with filepath.open(mode="w") as fp:
             yaml.dump(yaml_config, fp)
 
     def write_config_to_yaml(self) -> None:
-        """
-        Write a particular key/value to the desired configuration.yaml.
-
-        Example:
-        ```
-            [section]
-            - item1
-            - item2
-        ```
-
-        :param section: Section within the YAML file where the key/value will be recorded
-        :param values: List of items to be written to the YAML file
-        """
-        yaml_config = self.from_yaml_to_dict()
-        yaml_config["connections"] = self.connections
+        """Write Config instance's key-values to respective environment specific and global configuration.yml."""
```

Comment:
Can you write more description here? Just reading this function doc I'm not sure what this function is meant to do. Would like to know why we do this.

Reply:
Added more description.